code/game/machinery/weathercontrol.dm

```diff
@@ -0,0 +1,183 @@
+/obj/machinery/weathercontrol
+	name = "weather control device"
+	desc = "A device which uses weather control techniques such as cloud seeding to manipulate atmospheric conditions. It runs on bluespace crystals."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "wishgranter"
+	use_power = 1
+	density = 1
+	anchored = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK
+	req_access = list(access_rd)
+	idle_power_usage = 1500
+	var/error_message = null
+	var/efficiency_modifier = 0 //Subtract from crystal costs
+	var/crystal_reserve = 0 //Powered by bluespace crystals, this represents how much.
+
+/obj/machinery/weathercontrol/New()
+	. = ..()
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/weathercontrol,
+		/obj/item/weapon/cell/rad,
+		/obj/item/weapon/stock_parts/matter_bin/adv/super/bluespace,
+		/obj/item/weapon/stock_parts/micro_laser/high/ultra,
+		/obj/item/weapon/stock_parts/micro_laser/high/ultra)
+	RefreshParts()
+
+/obj/machinery/weathercontrol/Destroy()
+	if(crystal_reserve >= 300)
```

Comment:
300 ? Sounds like it would cause a lot of non-flawless crystals when destroyed. Is it a typo?

Reply:
At some point someone decided a flawless was worth 300. This only really matters if they get a flawless from spessmart and decide to use it here instead of on a tunneler or whatever.